src/core/execute.c

```diff
@@ -4909,6 +4909,19 @@ static int exec_child(
                         return r;
         }
 
+	if (context->memory_ksm) {
+#ifdef PR_SET_MEMORY_MERGE
+		r = prctl(PR_SET_MEMORY_MERGE, 1);
+		if (r < 0) {
+			*exit_status = EXIT_KSM;
+			return log_unit_error_errno(unit, r, "Failed to enable KSM %m");
```

Comment:
the second argument should be `errno`, not `r`, as prctl() returns errors via usual libc errno

Reply:
I'll change r to errno in the next version.